beta-src/src/components/ui/WDMain.tsx

```diff
@@ -253,6 +258,78 @@ const WDMain: React.FC = function (): React.ReactElement {
       isLivePhase: false,
     };
   };
+
+  useEffect(() => {
+    Object.entries(ordersMeta).forEach(([_, orderMeta]) => {
+      // Let's use as an example ION C APU - TUN
+      // The board should automatically also enters APU - TUN (Via Convoy)
+      // More info in CHR-68
+      const orderMetaUpdate = orderMeta.update;
+
+      if (
+        orderMetaUpdate?.type === "Convoy" &&
+        orderMetaUpdate?.fromTerrID &&
+        orderMetaUpdate.toTerrID
+      ) {
+        const fromTerr = maps.terrIDToProvince[orderMetaUpdate.fromTerrID];
+        const toTerr = maps.terrIDToProvince[orderMetaUpdate.toTerrID];
+
+        const legalConvoysByUnitIDs = Object.keys(
+          legalOrders.hasAnyLegalConvoysByUnitID,
+        );
+        if (legalConvoysByUnitIDs.length === 1) {
+          const availableViaConvoy = legalConvoysByUnitIDs[0];
+          // Verify if the order APU -> TUN is possible
+          const legalOrderViaConvoy =
+            legalOrders.legalConvoysByUnitID[availableViaConvoy];
+          if (legalOrderViaConvoy[fromTerr][toTerr]) {
+            // Get the unit ID that is located at the source territory (APU)
+            const asArray = Object.entries(gameDataResponse.data.units);
+            const findUnit = asArray.find(
+              (elem) =>
+                elem["1"].countryID === String(status.countryID) &&
+                elem["1"].terrID === orderMetaUpdate.fromTerrID &&
+                elem["1"].type === "Army",
+            );
+            if (findUnit && gameDataResponse.data.currentOrders) {
+              const unitID = findUnit["1"].id;
+              // Get the orderID related to the unit that needs to do Via Convoy, so e can update it.
+              const findOrder = gameDataResponse.data.currentOrders.find(
+                (currentOrder) => currentOrder.unitID === unitID,
+              );
+              const availableOrderID = findOrder?.id || "";
```

Comment:
What is the situation where findOrder is not found? And is it still correct to do the update in that situation, while putting in empty string as the id??

Reply:
Updated